tests/system/I18n/TimeTest.php

```diff
@@ -277,15 +277,27 @@ public function testCreateFromTimestamp(): void
 
         $timestamp = strtotime('2017-03-18 midnight');
 
+        // The timezone will be UTC if you don't specify.
         $time = Time::createFromTimestamp($timestamp);
 
-        $this->assertSame('Asia/Tokyo', $time->getTimezone()->getName());
-        $this->assertSame('2017-03-18 00:00:00', $time->format('Y-m-d H:i:s'));
+        $this->assertSame('UTC', $time->getTimezone()->getName());
+        $this->assertSame('2017-03-17 15:00:00', $time->format('Y-m-d H:i:s'));
 
         // Restore timezone.
         date_default_timezone_set($tz);
     }
 
+    public function testCreateFromTimestampWithMicrotime(): void
+    {
+        $timestamp = 1489762800.654321;
+
+        // The timezone will be UTC if you don't specify.
+        $time = Time::createFromTimestamp($timestamp);
+
+        // float cannot handle the number precisely.
+        $this->assertSame('2017-03-17 15:00:00.654300', $time->format('Y-m-d H:i:s.u'));
```

Comment:
I thought this was due to `float`, but the following test passes. Why?

```php
        $timestamp = 1489762800.654321;

        $time = \DateTimeImmutable::createFromTimestamp($timestamp);

        $this->assertSame('2017-03-17 15:00:00.654321', $time->format('Y-m-d H:i:s.u'));
```

Reply:
I meant the following test passes. Why `654300` and `654321`?
```php
        $timestamp = 1489762800.654321;

        $time1 = Time::createFromTimestamp($timestamp);
        $time2 = \DateTimeImmutable::createFromTimestamp($timestamp);

        $this->assertSame('2017-03-17 15:00:00.654300', $time1->format('Y-m-d H:i:s.u'));
        $this->assertSame('2017-03-17 15:00:00.654321', $time2->format('Y-m-d H:i:s.u'));
```

Reply:
It probably has something to do with the default `precision` (which is `14` by default).

In the `Time::createFromTimestamp` method, we use:

```php
$time = new static('@' . $timestamp, 'UTC', $locale);
```
Which will "cut" the ending and make it: `.6543`. We can either set `ini_set('precision', 16)` at the framework level which is not ideal or use `sprintf` here:

```php
$time = new static(sprintf('@%.6f', $timestamp), 'UTC', $locale);
```

Reply:
`sprintf('@%.6f', $timestamp)` works. Thank you!